### PR TITLE
test: /postgres engine spec test coverage

### PR DESCRIPTION
**Pull request by kstrz**
### SUMMARY
Increase postgres engine spec test coverage.
Fix estimate_statement_cost for invalid syntax

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/db_engine_specs/postgres.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 from datetime import datetime
 from typing import Any, Dict, List, Optional, Tuple, TYPE_CHECKING
 
+from psycopg2 import errors
```
Comment (villebro) [inline superset/db_engine_specs/postgres.py:21]: We need to move this import into the method, as `psycopg2` isn't a required dependency.

File: tests/db_engine_specs/postgres_tests.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 # under the License.
 from unittest import mock
 
+from psycopg2 import errors
```
Comment (villebro) [inline tests/db_engine_specs/postgres_tests.py:19]: same here

**Review by villebro — COMMENTED** [review]
A few quick first pass comments

File: superset/db_engine_specs/postgres.py (hunk 1 above)
Comment (kstrz) [reply superset/db_engine_specs/postgres.py]: done

File: tests/db_engine_specs/postgres_tests.py (hunk 2 above)
Comment (kstrz) [reply tests/db_engine_specs/postgres_tests.py]: done

**Comment by kstrz** [comment]
Hey! Could you look at it? @willbarrett @dpgaspar

File: superset/db_engine_specs/postgres.py [hunk 3]
```diff
@@ -78,8 +78,13 @@ def get_allow_cost_estimate(cls, extra: Dict[str, Any]) -> bool:
 
     @classmethod
     def estimate_statement_cost(cls, statement: str, cursor: Any) -> Dict[str, Any]:
+        from psycopg2 import errors
+
         sql = f"EXPLAIN {statement}"
-        cursor.execute(sql)
+        try:
+            cursor.execute(sql)
+        except errors.SyntaxError as err:
```
Comment (kstrz) [inline superset/db_engine_specs/postgres.py:86]: I added handling exception but now I'm wondering if leaving it to be caught by errorhandler would be a better option.

Comment (dpgaspar) [reply superset/db_engine_specs/postgres.py:86]: I think so too

**Review by dpgaspar — COMMENTED** [review]
nit: can you post the current coverage and the new coverage after the PR?

File: superset/db_engine_specs/postgres.py (hunk 3 above)
Comment (kstrz) [reply superset/db_engine_specs/postgres.py]: done

**Comment by villebro** [comment]
@kstrz this needs to be rebased

**Comment by kstrz** [comment]
Coverage for superset/db_engine_specs/postgres.py on master: 85.96% , on current branch: 96.05%

**Review by villebro — APPROVED** [review]

**Pull request merged**